utils.cpp

```diff
@@ -245,66 +260,26 @@ HPHP::Object Utils::doExecuteCommand(const char *db, mongoc_client_t *client, in
 		cursor->server_id = server_id;
 	}
 
-	if (!mongoc_cursor_next(cursor, &doc)) {
-		bson_error_t error;
+	/* This throws an exception upon error */
+	hippo_advance_cursor_and_check_for_error(cursor);
 
-		if (mongoc_cursor_error(cursor, &error)) {
-			mongoc_cursor_destroy(cursor);
-			throw Utils::throwExceptionFromBsonError(&error);
-		}
-	}
+	if (bson_iter_init_find(&iter, mongoc_cursor_current(cursor), "cursor") && BSON_ITER_HOLDS_DOCUMENT(&iter)) {
+		mongoc_cursor_t *cmd_cursor;
 
-	/* This code is copied from phongo, which has it adapated from
-	 * _mongoc_cursor_cursorid_prime(), but we avoid * advancing the cursor,
-	 * since we are already positioned at the first result * after the error
-	 * checking above. */
-	if (bson_iter_init_find(&iter, doc, "cursor") && BSON_ITER_HOLDS_DOCUMENT(&iter) && bson_iter_recurse(&iter, &child)) {
-		mongoc_cursor_cursorid_t *cid;
-		bson_t empty = BSON_INITIALIZER;
-
-		_mongoc_cursor_cursorid_init(cursor, &empty);
-		cursor->limit = 0;
-
-		cid = (mongoc_cursor_cursorid_t*) cursor->iface_data;
-		cid->in_batch = true;
-		bson_destroy(&empty);
-
-		while (bson_iter_next(&child)) {
-			if (BSON_ITER_IS_KEY(&child, "id")) {
-				cursor->rpc.reply.cursor_id = bson_iter_as_int64(&child);
-			} else if (BSON_ITER_IS_KEY(&child, "ns")) {
-				const char *ns;
-
-				ns = bson_iter_utf8(&child, &cursor->nslen);
-				bson_strncpy(cursor->ns, ns, sizeof cursor->ns);
-			} else if (BSON_ITER_IS_KEY(&child, "firstBatch")) {
-				if (BSON_ITER_HOLDS_ARRAY(&child) && bson_iter_recurse(&child, &cid->batch_iter)) {
-					cid->in_batch = true;
-				}
-			}
-		}
+		/* According to mongoc_cursor_new_from_command_reply(), the reply bson_t
+		 * is ultimately destroyed on both success and failure. Use bson_copy()
+		 * to create a writable copy of the const bson_t we fetched above. */
+		cmd_cursor = mongoc_cursor_new_from_command_reply(client, bson_copy(mongoc_cursor_current(cursor)), mongoc_cursor_get_hint(cursor));
```

Comment:
How so? This PR is a single commit, so you should be able to just reset one commit back and ignore the irrelevant changes when recreating the commit.